docs/tuning-apps/mpi-io/ompio.rst

```diff
@@ -0,0 +1,262 @@
+
+   shell$ mpirun --mca fcoll dynamic -n 64 ./a.out
+
+``fs`` and ``fbtl`` components are typically chosen based on the file
+system type utilized (e.g. the ``pvfs2`` component is chosen when the
+file is located on an PVFS2/OrangeFS file system, the ``lustre``
+component is chosen for Lustre file systems, etc.). The ``ufs`` ``fs``
+component is used if no file system specific component is availabe
+(e.g. local file systems, NFS, BeefFS, etc.), and the ``posix``
+``fbtl`` component is used as the default component for read/write
+operations.
+
+The ``fcoll`` framework provides several different components. The
+current decision logic in OMPIO uses the file view provided by the
+application as well as file system level characteristics (e.g. file
+system, stripe width) to determine which component to use. The most
+important ``fcoll`` components are:
+
+* ``dynamic_gen2``: the default component used on lustre file
+  system. This component is based on the two-phase I/O algorithm with
+  a static file partioning strategy, i.e. an aggregator processes will
+  by default only write data to a single storage server.
+
+* ``vulcan``: the default component used on all other file
+  systems. This component is based on the two-phase I/O algorithm with
+  an even file partitioning strategy, i.e. each of the n aggregators
+  will write 1/n th of the overall file.
+
+* ``individual``: this components executes all collective I/O
+  operations in terms of individual I/O operations.
+
+The ``sharedfp`` framework provides a different implementation of the
+shared file pointer operations depending on file system features.
+
+* ``lockfile``: this component will be used on file system which
+  support for file locking.
+
+* ``sm``: component used in scenarios in which all processes of the
+  communicator are on the same physical node.
+
+* ``individual``: a component that can be used if neither of the other
+  two components are available. This component provides however only
+  limited functionality (i.e. write operations only).
+
+  .. note:: See :ref:`the section on the individual sharedfp component
+            <label-ompio-individual-sharedfp>` to understand
+            functionality and limitations.
+
+Tuning OMPIO performance
+------------------------
+
+The most important parameters influencing the performance of an I/O
+operation are listed below:
+
+#. ``io_ompio_cycle_buffer_size``: Data size issued by individual
+   reads/writes per call. By default, an individual read/write
+   operation will be executed as one chunk. Splitting the operation up
+   into multiple, smaller chunks can lead to performance improvements
+   in certain scenarios.
+
+#. ``io_ompio_bytes_per_agg``: Size of temporary buffer for collective
+   I/O operations on aggregator processes. Default value is 32MB.
+   Tuning this parameter has a very high impact on the performance of
+   collective operations.
+
+   .. note:: Be sure to also see recommendations for tuning collective
```

Comment:
@edgargabriel What section are you referring to here?  We should put in a `:ref:` cross reference.

Reply:
will have to remove that, we used to have separate section on tuning collectives, but I decided to remove it, was more confusing than helpful